src/io/inputs.jl

```diff
@@ -121,76 +122,65 @@ end
         )
-    end
-    dim_labels[i] = sites
 
-    try
-        loaded = NamedDimsArray(data; zip(dim_names, dim_labels)...)
-    catch err
-        if isa(err, KeyError)
-            n_sites = size(data, 2)
-            @warn "Provided file $(data_fn) did not have the expected dimensions (one of: timesteps, reef_siteid, scenarios)."
-            if n_sites != nrow(site_data)
-                error(
-                    "Mismatch in number of sites ($(data_fn)). Expected $(nrow(site_data)), got $(n_sites)",
-                )
+        try
+            loaded_data = NamedDimsArray(data; zip(dim_names, dim_labels)...)
+        catch err
+            if isa(err, KeyError)
+                n_sites = size(data, 2)
+                @warn "Provided file $(data_fn) did not have the expected dimensions " *
+                    "(one of: timesteps, reef_siteid, scenarios)."
+                if n_sites != nrow(site_data)
+                    error(
+                        "Mismatch in number of sites ($(data_fn)). " *
+                        "Expected $(nrow(site_data)), got $(n_sites)",
+                    )
+                end
+            else
+                rethrow(err)
             end
-        else
-            rethrow(err)
         end
     end
 
-    return loaded
+    return loaded_data
 end
 
 """
-    _char_to_string(vals)::Vector{String}
+Return vector of labels for each array dimension. Important: Cannot trust indicated
+dimension metadata to get site labels, because this can be incorrect. Instead, match by
+number of sites.
+"""
+function _nc_dim_labels(
+    data_fn::String, data::Array{<:AbstractFloat}, nc_file::NetCDF.NcFile
+)::Vector{Union{UnitRange{Int64},Vector{String}}}
+    local sites_idx::Int64
 
-Convert character array entries in netCDFs to string.
+    has_reef_siteid_var = "reef_siteid" in keys(nc_file.vars)
+    sites = has_reef_siteid_var ? NetCDF.readvar(nc_file, "reef_siteid") : 1:size(data, 2)
 
-Some packages used to write out netCDFs do not yet support
-string values, and instead reverts to writing out character arrays.
-"""
-function _char_to_string(vals::AbstractVecOrMat)::Vector{String}
-    if vals isa Matrix
-        vals = map(x -> join(skipmissing(x)), eachcol(vals))
+    try
+        # This will be an issue if two or more dimensions have the same number of elements
+        # as the number of sites, but so far it hasn't happened...
+        sites_idx = first(findall(size(data) .== length(sites)))
+    catch err
+        error(
+            "Error loading $data_fn : could not determine number of locations." *
+            "Detected size: $(size(data)) | Known number of locations: $(length(sites))",
+        )
     end
 
-    # R's ncdf4 package does not yet support string values
-    # so strip the null terminator from the joined character array.
-    vals = replace.(vals, "\0" => "")
+    dim_labels = Union{UnitRange{Int64},Vector{String}}[1:n for n in size(data)]
+    dim_labels[sites_idx] = sites
```

Comment:
How about:


```suggestion
    dim_labels = Union{UnitRange{Int64},Vector{String}}[
        idx != sites_idx ? 1:n : sites for (idx, n) in enumerate(size(data))
    ]
```

Reply:
The current approach is probably more readable though

Reply:
Yes... I thought about doing what you wrote, but it's harder to read and I didn't see any big advantage... But I'm ok with changing, what do you think?

Reply:
Benefit of doing it in a single pass is small, I'm okay with leaving it as is.